File: CodeForces/977b/main.cpp
#include <iostream>
#include <unordered_map>

using namespace std;

int main() 
{

    int n;
    string s;
    cin >> n >> s;

    int maxi = -1;
    string ans = "";
    unordered_map<string, int> mp;
    
    for(int i = 0; i < n - 1; i++) {
        string ss = s.substr(i, 2);
        mp[ss]++;
        if(mp[ss] >= maxi) {
            maxi = mp[ss];
            ans = ss;
        }
    }   

    cout << ans << "\n";

    return 0;
}

File: CodeForces/977b/main.py
from collections import defaultdict


def inp():
    return int(input())


def inlt():
    return list(map(int, input().split()))


def insr():
    return input().strip()


def solve():
    n = inp()
    arr = insr()
    mp = defaultdict(int)

    ans = ""
    maxi = -1

    for i in range(n):
        substr = arr[i : i + 2]
        if len(substr) == 2:
            mp[substr] += 1
            if mp[substr] >= maxi:
                maxi = mp[substr]
                ans = substr

    print(ans)
    return


solve()
